File: services/inventory-service/src/main/java/io/minishop/inventory/web/dto/InventoryResponse.java
package io.minishop.inventory.web.dto;

import io.minishop.inventory.domain.Inventory;

public record InventoryResponse(
        Long productId,
        Integer availableQuantity
) {
    public static InventoryResponse from(Inventory inv) {
        return new InventoryResponse(inv.getProductId(), inv.getAvailableQuantity());
    }
}

File: services/inventory-service/src/main/kotlin/io/minishop/inventory/web/dto/InventoryResponse.kt
package io.minishop.inventory.web.dto

import io.minishop.inventory.domain.Inventory

@JvmRecord
data class InventoryResponse(
    val productId: Long?,
    val availableQuantity: Int?,
) {
    companion object {
        @JvmStatic
        fun from(inv: Inventory): InventoryResponse =
            InventoryResponse(inv.productId, inv.availableQuantity)
    }
}
